### PR TITLE
TPM2 encoding cleanup

### PR DESCRIPTION
**Pull request by awly**
This makes encoding/decoding funcs in tpm2.go more readable, documented and fixes a few bugs.

### CODE REVIEW

File: tpm2/tpm2.go [hunk 1]
```diff
@@ -216,79 +63,112 @@ func encodeShortPCRs(pcrNums []int) ([]byte, error) {
...
 	// size(3)  mask     mask     mask
 	// 00000011 00000000 00000001 00000100
-	pcr := make([]byte, sizeOfPCRSelect+1)
-	pcr[0] = sizeOfPCRSelect
+	ts := tpmsPCRSelection{
+		Hash: sel.Hash,
+		Size: sizeOfPCRSelect,
+		PCRs: make([]byte, sizeOfPCRSelect),
+	}
 	// pcrNums parameter is indexes of PCRs, convert that to set bits.
```
Comment (tmroeder) [inline tpm2/tpm2.go]: nit: I think this comment has gone stale, since there isn't a pcrNums parameter any more.

**Comment by tmroeder** [comment]
This looks great, thanks. I only have one tiny comment fix to suggest.

File: tpm2/tpm2.go (hunk 1 above)
Comment (awly) [reply tpm2/tpm2.go]: Done

**Comment by awly** [comment]
Thanks for the quick review!

**Pull request merged**